tox.ini

```diff
@@ -1,68 +1,41 @@
 [flake8]
-max-line-length=170
 exclude=connexion/__init__.py
-rst-roles=class
+rst-roles=class,mod,obj
+# https://black.readthedocs.io/en/stable/guides/using_black_with_other_tools.html#flake8
+# Longest docstring in current code base
+max-line-length=137
+extend-ignore=E203,RST303
 
 [tox]
+isolated_build = True
 envlist =
-    {py37}-{min,pypi,dev}
-    {py38}-{min,pypi,dev}
-    {py39}-{min,pypi,dev}
-    isort-check
-    isort-check-examples
-    isort-check-tests
-    flake8
-    mypy
+    {py39,py310,py311,py312}-pypi
+    pre-commit
 
 [gh-actions]
 python =
-    3.7: py37-min,py37-pypi
-    3.8: py38-min,py38-pypi
-    3.9: py39-min,py39-pypi,flake8,isort-check,isort-check-examples,isort-check-tests,mypy
+    3.9: py39-pypi
+    3.10: py310-pypi
+    3.11: py311-pypi,pre-commit
+    3.12: py312-pypi
```

Comment:
The [github actions workflow](https://github.com/spec-first/connexion/blob/3d67f16075349d7137c2848aef9fa3bd35e3a268/.github/workflows/pipeline.yml) should be updated to match this.